projects/igniteui-angular/src/lib/grids/grid/cell.spec.ts

```diff
@@ -1,541 +1,391 @@
-        const event = new Event('dblclick');
-        spyOn(event, 'preventDefault');
-        cellElem.nativeElement.dispatchEvent(event);
-        const args: IGridCellEventArgs = {
-            cell: firstCell,
-            event
-        };
-
-        fix.detectChanges();
-
-        expect(event.preventDefault).not.toHaveBeenCalled();
-        expect(grid.onDoubleClick.emit).toHaveBeenCalledWith(args);
-        expect(firstCell).toBe(fix.componentInstance.clickedCell);
-    });
-
-    it('Should not attach doubletap handler for non-iOS', () => {
-        const addListenerSpy = spyOn(HammerGesturesManager.prototype, 'addEventListener');
-        const platformUtil: PlatformUtil = TestBed.get(PlatformUtil);
-        const oldIsIOS = platformUtil.isIOS;
-        platformUtil.isIOS = false;
-        const fix = TestBed.createComponent(DefaultGridComponent);
-        fix.detectChanges();
-        // spyOnProperty(PlatformUtil.prototype, 'isIOS').and.returnValue(false);
-        expect(addListenerSpy).not.toHaveBeenCalled();
-
-        platformUtil.isIOS = oldIsIOS;
-    });
+    describe('Test events', () => {
+        let fix;
+        let grid: IgxGridComponent;
+
+        configureTestSuite();
+        beforeAll(async(() => {
+            TestBed.configureTestingModule({
+                declarations: [
+                    NoScrollsComponent
+                ],
+                imports: [NoopAnimationsModule, IgxGridModule],
+            }).compileComponents();
+        }));
+
+        beforeEach(fakeAsync(/** height/width setter rAF */() => {
+            fix = TestBed.createComponent(NoScrollsComponent);
+            fix.detectChanges();
+            grid = fix.componentInstance.grid;
+        }));
+
+        it('@Input properties and getters', () => {
+            const firstIndexCell = grid.getCellByColumn(0, 'ID');
+
+            expect(firstIndexCell.columnIndex).toEqual(grid.columnList.first.index);
+            expect(firstIndexCell.rowIndex).toEqual(grid.rowList.first.index);
+            expect(firstIndexCell.grid).toBe(grid);
+            expect(firstIndexCell.nativeElement).toBeDefined();
+            expect(firstIndexCell.nativeElement.textContent).toMatch('1');
+            expect(firstIndexCell.readonly).toBe(true);
+            expect(firstIndexCell.describedby).toMatch(`${grid.id}_${firstIndexCell.column.field}`);
+        });
 
-    it('Should handle doubletap on iOS, trigger onDoubleClick event', () => {
-        const addListenerSpy = spyOn(HammerGesturesManager.prototype, 'addEventListener');
-        const platformUtil: PlatformUtil = TestBed.get(PlatformUtil);
-        const oldIsIOS = platformUtil.isIOS;
-        platformUtil.isIOS = true;
-        const fix = TestBed.createComponent(DefaultGridComponent);
-        fix.detectChanges();
+        it('selection and selection events', () => {
+            const rv = GridFunctions.getRowCells(fix, 0)[0];
+            const cell = grid.getCellByColumn(0, 'ID');
```

Comment:
I see that getting `GridFunctions.getRowCells(fix, 0)[0]` and `grid.getCellByColumn(0, 'ID')`  is done in every test in this describe, so this could also be moved into the beforeEach.